File: backend-api-dot-net/KPW.Domain/Entities/OwnerSubjectiveNote.cs
using KPW.Domain.Common;

namespace KPW.Domain.Entities;

public class OwnerSubjectiveNote : AuditableEntity
{
    public int OwnerSubjectiveNoteId { get; set; }
    public int PetId { get; set; }
    public int OwnerId { get; set; }
    public DateTime NoteDate { get; set; } = DateTime.UtcNow;

    public string Notes { get; set; } = string.Empty;
    public int? PainObserved { get; set; }
    public int? EnergyObserved { get; set; }
    public bool IsReviewed { get; set; }

    public Pet Pet { get; set; } = null!;
    public User Owner { get; set; } = null!;
}

File: owner-app-flutter/lib/features/pets/models/owner_subjective_note.dart
class OwnerSubjectiveNote {
  const OwnerSubjectiveNote({
    required this.ownerSubjectiveNoteId,
    required this.petId,
    required this.ownerId,
    required this.ownerName,
    required this.noteDate,
    required this.notes,
    this.painObserved,
    this.energyObserved,
    required this.isReviewed,
  });

  final int ownerSubjectiveNoteId;
  final int petId;
  final int ownerId;
  final String ownerName;
  final DateTime noteDate;
  final String notes;
  final int? painObserved;
  final int? energyObserved;
  final bool isReviewed;

  factory OwnerSubjectiveNote.fromJson(Map<String, dynamic> json) {
    return OwnerSubjectiveNote(
      ownerSubjectiveNoteId: json['ownerSubjectiveNoteId'] as int? ?? 0,
      petId: json['petId'] as int? ?? 0,
      ownerId: json['ownerId'] as int? ?? 0,
      ownerName: json['ownerName'] as String? ?? 'Owner',
      noteDate: json['noteDate'] != null
          ? DateTime.parse(json['noteDate'] as String)
          : DateTime.now(),
      notes: json['notes'] as String? ?? '',
      painObserved: json['painObserved'] as int?,
      energyObserved: json['energyObserved'] as int?,
      isReviewed: json['isReviewed'] as bool? ?? false,
    );
  }
}
